spec/models/event_spec.rb

```diff
@@ -41,4 +41,30 @@
       it { is_expected.to_not be_valid }
     end
   end
+
+  describe '.requiring_notification' do
+    context 'when there are events alrigning with the NOTIFICATION_PERIOD' do
+      let(:required_notif) do
+        FactoryGirl.create(
+          :event, date: Time.zone.today - Event::NOTIFICATION_PERIOD
+        )
+      end
+      let(:unrequired_notif) do
+        FactoryGirl.create(
+          :event, date: Time.zone.today + Event::NOTIFICATION_PERIOD
+        )
+      end
+      it 'returns an array of Events requiring notification' do
+        expect(Event.requiring_notification).to include(required_notif)
+      end
+      it 'doesnt include Events that arent requiring notification' do
+        expect(Event.requiring_notification).not_to include(unrequired_notif)
+      end
+    end
+    context 'when there are no events aligning with the NOTIFICATION_PERIOD' do
+      it 'returns an empty array' do
+        expect(Event.requiring_notification.empty?).to eq(true)
```

Comment:
Rspec allows you to dynamically use predicate methods with the `be` helper. 

`expect([]).to be_empty`